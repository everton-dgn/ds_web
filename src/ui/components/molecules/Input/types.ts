import { InputHTMLAttributes } from 'react'

export type InputProps = InputHTMLAttributes<HTMLInputElement> & {
  label?: string
  error?: string
}
