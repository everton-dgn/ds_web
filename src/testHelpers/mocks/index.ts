export * from './matchMedia'
export * from './reactRouterDom'
