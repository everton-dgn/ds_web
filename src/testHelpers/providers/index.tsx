export * from './providerComponents'
export * from './providerHooks'
