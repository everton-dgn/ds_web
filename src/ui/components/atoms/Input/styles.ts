import styled from 'styled-components'
import T from 'ui/theme'

export const Input = styled.input`
  width: 100%;
  height: 50px;
  border-radius: ${T.borderRadius.s};
  padding: 5px 20px;
  border: 2px solid transparent;
  box-shadow: 0 0 0 1px #333;
  transition: 0.1s linear;

  &:focus {
    border: 2px solid #222;
    box-shadow: 0 0 0 2px #fff;
  }
`
