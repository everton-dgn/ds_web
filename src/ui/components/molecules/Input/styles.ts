import styled from 'styled-components'
import T from 'ui/theme'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const Label = styled.label`
  display: flex;
  margin-bottom: 5px;
`

export const WrapperError = styled.span`
  display: flex;
  width: 100%;
  min-height: 16px;
  margin-top: 8px;
`

export const Input = styled.input<{ error?: string }>`
  display: flex;
  width: 100%;
  font-size: ${T.fonts.sizes.paragraph};
  font-weight: ${T.fonts.weights.regular};
  color: ${({ error }) => (error ? T.colors.error : T.colors.black)};
  height: 39px;
  border-radius: ${T.borderRadius.s};
  padding: 0 14px;
  border: 1px solid ${({ error }) => (error ? T.colors.error : T.colors.black)};
  transition: 0.1s linear;

  &::placeholder {
    color: ${T.colors.grey};
  }

  &:focus,
  &:active {
    border: 1px solid
      ${({ error }) => (error ? T.colors.error : T.colors.black)};
  }
`
