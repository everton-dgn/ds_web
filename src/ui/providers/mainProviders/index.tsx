import { GlobalStyles } from 'ui/globalStyles'
import * as C from 'ui/components'
import { MainProvidersProps } from './types'

export const MainProviders = ({ children }: MainProvidersProps) => (
  <C.ErrorBoundary>
    {children}
    <GlobalStyles />
  </C.ErrorBoundary>
)
