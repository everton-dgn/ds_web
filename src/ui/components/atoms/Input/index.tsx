import { forwardRef, memo } from 'react'
import * as S from './styles'
import { InputProps } from './types'

const Input = forwardRef<HTMLInputElement, InputProps>(
  ({ type = 'text', ...props }, ref) => (
    <S.Input ref={ref} {...props} type={type} />
  )
)

export default memo(Input)
