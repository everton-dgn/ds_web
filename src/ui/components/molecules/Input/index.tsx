import { forwardRef, memo } from 'react'
import * as C from 'ui/components'
import * as S from './styles'
import { InputProps } from './types'

const Input = forwardRef<HTMLInputElement, InputProps>(
  ({ type = 'text', label, error, ...props }, ref) => (
    <S.Wrapper>
      {label && (
        <S.WrapperLabel>
          <C.Typography text={label} type="paragraphHighlight" align="left" />
        </S.WrapperLabel>
      )}
      <S.Input ref={ref} {...props} type={type} error={error} />
      <S.WrapperError>
        {error && (
          <C.Typography
            text={error}
            type="caption"
            align="left"
            color="error"
          />
        )}
      </S.WrapperError>
    </S.Wrapper>
  )
)

export default memo(Input)
