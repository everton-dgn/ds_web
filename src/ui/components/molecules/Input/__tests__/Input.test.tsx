import { screen } from '@testing-library/react'
import { renderWithProviders } from 'testHelpers/providers'
import Input from '..'

describe('[Component] Input', () => {
  it('should render a snapshot', () => {
    const { container } = renderWithProviders(
      <Input
        name="searchRepositoriesInTheGithub"
        placeholder="Nome de usuário"
        onSetField={jest.fn()}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render the Input', () => {
    const onSetField = jest.fn()
    renderWithProviders(
      <Input
        name="searchRepositoriesInTheGithub"
        placeholder="Nome de usuário"
        onSetField={onSetField}
      />
    )
    const wrapper = screen.getByRole('textbox')
    expect(wrapper).toBeInTheDocument()
  })
})
