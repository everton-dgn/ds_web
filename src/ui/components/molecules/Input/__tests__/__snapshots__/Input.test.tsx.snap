// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Input should render a snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  min-height: 16px;
  margin-top: 8px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  font-size: 14px;
  font-weight: 400;
  color: #282828;
  height: 39px;
  border-radius: 2px;
  padding: 0 14px;
  border: 1px solid #282828;
  -webkit-transition: 0.1s linear;
  transition: 0.1s linear;
}

.c1::-webkit-input-placeholder {
  color: #d0d0d0;
}

.c1::-moz-placeholder {
  color: #d0d0d0;
}

.c1:-ms-input-placeholder {
  color: #d0d0d0;
}

.c1::placeholder {
  color: #d0d0d0;
}

.c1:focus,
.c1:active {
  border: 1px solid #282828;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      name="searchRepositoriesInTheGithub"
      placeholder="Nome de usuÃ¡rio"
      type="text"
    />
    <span
      class="c2"
    />
  </div>
</div>
`;
